packages/kit/src/runtime/server/page/render.js

```diff
@@ -353,17 +354,34 @@ export async function render_response({
 	// add the content after the script/css links so the link elements are parsed first
 	head += rendered.head;
 
+	const html = options.app_template({
+		head,
+		body,
+		assets,
+		nonce: /** @type {string} */ (csp.nonce)
+	});
+
 	// TODO flush chunks as early as we can
-	const html =
+	const transformed =
 		(await resolve_opts.transformPageChunk({
-			html: options.app_template({ head, body, assets, nonce: /** @type {string} */ (csp.nonce) }),
+			html,
 			done: true
 		})) || '';
 
+	if (DEV) {
+		if (transformed.split('<!--').length < html.split('<!--').length) {
+			// the \u001B stuff is ANSI codes, so that we don't need to add a library to the runtime
+			// https://svelte.dev/repl/1b3f49696f0c44c881c34587f2537aa2
+			console.warn(
```

Comment:
I think we should emit this warning only if `csr` is `true` (as mentioned in the linked issue), since if we don't need hydration if it's turned off (there's no JS delivered to the page)